geopy/geocoders/uscensus.py

```diff
@@ -0,0 +1,189 @@
+from functools import partial
+from urllib.parse import urlencode
+
+from geopy.geocoders.base import DEFAULT_SENTINEL, Geocoder
+from geopy.location import Location
+from geopy.point import Point
+from geopy.util import logger
+
+__all__ = ('USCensus', )
+
+
+class USCensus(Geocoder):
+    """ Geocoder that uses the United States Census Bureau geocode API
+
+    Documentation at:
+        https://geocoding.geo.census.gov/geocoder
+    """
+
+    structured_query_params = {
+        'street',
+        'city',
+        'state',
+        'zip',
+    }
+
+    def __init__(
+        self,
+        *,
+        timeout=DEFAULT_SENTINEL,
+        proxies=DEFAULT_SENTINEL,
+        domain='geocoding.geo.census.gov',
+        scheme=None,
+        user_agent=None,
+        ssl_context=DEFAULT_SENTINEL,
+        adapter_factory=None
+    ):
+        """
+        :param int timeout:
+            See :attr:`geopy.geocoders.options.default_timeout`.
+
+        :param dict proxies:
+            See :attr:`geopy.geocoders.options.default_proxies`.
+
+        :param str domain: Domain where the target US Census geocoder service
+            is hosted.
+
+        :param str scheme:
+            See :attr:`geopy.geocoders.options.default_scheme`.
+
+        :param str user_agent:
+            See :attr:`geopy.geocoders.options.default_user_agent`.
+
+        :type ssl_context: :class:`ssl.SSLContext`
+        :param ssl_context:
+            See :attr:`geopy.geocoders.options.default_ssl_context`.
+
+        :param callable adapter_factory:
+            See :attr:`geopy.geocoders.options.default_adapter_factory`.
+
+            .. versionadded:: 2.0
+        """
+        super().__init__(
+            scheme=scheme,
+            timeout=timeout,
+            proxies=proxies,
+            user_agent=user_agent,
+            ssl_context=ssl_context,
+            adapter_factory=adapter_factory
+        )
+        self.api = "%s://%s/geocoder" % (self.scheme, domain)
+
+    def geocode(
+        self,
+        query,
+        *,
+        dataset_type='Public_AR',
+        spatial_benchmark='Current',
+        geolookup=False,
+        geography_vintage='Current',
+        layers=None,
+        exactly_one=True,
+        timeout=16
```

Comment:
Geopy geocoders should have `DEFAULT_SENTINEL` value as default for `timeout`. I have just merged additional tests in #480 which would check this property. So if you choose to continue working on getting this merged, please rebase on the fresh master branch and fix the test errors.